### DISCUSSION TITLE
WebOne 0.11.0

### DISCUSSION THREAD

**Discussion by atauenis** [atauenis/webone#42] [General]
New stable release of WebOne, released on 14.05.2021 after 27 commits.

* Changes in configuration file(s) processing:
   * Added support of `[Include:path/filename.ext]` directives to load configuration from multiple files.
   * Configuration file(s) now may contain spaces and tabulations for easier readability. They won't affect work.
   * Default configuration file on Linux/macOS is now at `/etc/webone.conf`, not at strange `/etc/WebOne/webone.conf` location.
   * Linux and macOS users now may store own WebOne configuration files in `/etc/webone.conf.d/` directory. So package updates will not overwrite user configuration every time.
   * Windows users now may store own multiple WebOne configuration files in program directory or in `C:\ProgramData\WebOne\`, and they won't be overwritten on upgrades.
* Changed the executable file name case: `WebOne` -> `webone`. Now it is consistent with configuration file name and with most of applications.
* Output encoding now can be set via `AddOutputEncoding` edit rule in edit sets too. By default this is used in `codepage.conf` to allow pre-Unicode-era browsers displaying webpages originally served in Unicode. Currently there is auto-detection of WinMosaic, Netscape <4 (Windows, MacOS 7-9), MSIE <4, DOSlynx, xChaos Arachne.
* Alphabet transliteration now may be enabled/disabled in edit sets by using `AddTranslit=yes` edit rule. Just leaving `[Translit]` section of webone.conf uncommented is no longer enough now (this confused users previously).
* Added log rotation profile on Linux, so WebOne log file will no longer be huge after intensive use.
* Added ability to find and replace something in HTTP headers: use `AddResponseHeaderFind`, `AddResponseHeaderReplace` and `AddRequestHeaderFind`, `AddRequestHeaderReplace` editing rules.
* Added support for multiple user accounts on password protected servers. Use `[Authenticate]` section of configuration files.
* Added `AuthenticateMessage`, `AuthenticateRealm` options in `[Authenticate]` section(s) for configuring proxy credentials request window and error page.
* Added store of client's user name in server log if password protection is enabled.
* Fixed inability to download Proxy Automatic Configuration script by some browsers when the proxy is password protected.
* Fixed slow exiting by Ctrl+C or window close.
* Fixed display of error messages (pages) when the messages are non-English and the client browser is not UTF-aware.

**Documentation in wiki will be updated soon.**

<hr /><em>This discussion was created from the release <a href='https://github.com/atauenis/webone/releases/tag/v0.11.0'>WebOne 0.11.0</a>.</em>

**Comment by georgebohnisch** [comment]
Excellent! Thanks for implementing the user name in server logs.
